VotingApplication/VotingApplication.Web.Api.Tests/Controllers/ManageInvitationControllerTests.cs

```diff
@@ -0,0 +1,193 @@
+            var mockContext = new Mock<IVotingContext>();
+            mockContextFactory.Setup(a => a.CreateContext()).Returns(mockContext.Object);
+            mockContext.Setup(a => a.Polls).Returns(dummyPolls);
+
+            var mockMailSender = new Mock<IMailSender>();
+
+            _controller = new ManageInvitationController(mockContextFactory.Object, mockMailSender.Object);
+            _controller.Request = new HttpRequestMessage();
+            _controller.Configuration = new HttpConfiguration();
+        }
+
+        #region GET
+
+        [TestMethod]
+        public void GetIsNotAllowed()
+        {
+            // Act
+            var response = _controller.Get(_mainUUID);
+
+            // Assert
+            Assert.AreEqual(HttpStatusCode.MethodNotAllowed, response.StatusCode);
+        }
+
+        [TestMethod]
+        public void GetByIdIsNotAllowed()
+        {
+            // Act
+            var response = _controller.Get(_mainUUID, 1);
+
+            // Assert
+            Assert.AreEqual(HttpStatusCode.MethodNotAllowed, response.StatusCode);
+        }
+
+        #endregion
+
+        #region POST
+
+        [TestMethod]
+        public void PostIsAllowed()
+        {
+            // Act
+            var response = _controller.Post(_mainUUID, new List<string>());
+
+            // Assert
+            Assert.AreEqual(HttpStatusCode.OK, response.StatusCode);
+        }
+
+        [TestMethod]
+        public void PostWithInvalidPollIdIsRejected()
+        {
+            // Act
+            Guid newGuid = Guid.NewGuid();
+            var response = _controller.Post(newGuid, new List<string>());
+
+            // Assert
+            Assert.AreEqual(HttpStatusCode.NotFound, response.StatusCode);
+            HttpError error = ((ObjectContent)response.Content).Value as HttpError;
+            Assert.AreEqual("Poll " + newGuid + " not found", error.Message);
+        }
+
+        [TestMethod]
+        public void PostCreatesNewTokensForInviteOnlyPoll()
+        {
+            // Act
+            var invitations = new List<string>() { "a@b.c", "d@e.f" };
+            _controller.Post(_inviteOnlyUUID, invitations);
+
+            // Assert
+            Assert.AreEqual(2, _inviteOnlyPoll.Tokens.Count);
+        }
+
+        [TestMethod]
+        public void PostDoesNotCreatesNewTokensForOpenPoll()
+        {
+            // Act
+            _controller.Post(_mainUUID, new List<string>());
+
+            // Assert
+            Assert.AreEqual(0, _mainPoll.Tokens.Count);
+        }
+
+        [TestMethod]
+        public void PostDoesNotCreatesNewTokensForMalformedEmailsInInviteOnlyPoll()
+        {
+            // Act
+            var invitations = new List<string>() { "a@b.c", "d@e.f", @"notAnEmail", "also@notAnEmail", @"neither@is.this.", @"    but@this.is  " };
+            _controller.Post(_inviteOnlyUUID, invitations);
+
+            // Assert
+            Assert.AreEqual(3, _inviteOnlyPoll.Tokens.Count);
```

Comment:
If possible, it is probably worth checking that the correct 3 are accepted here.